externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -62,49 +63,49 @@ class ExecuteStatement(
     OperationLog.removeCurrentOperationLog()
   }
 
-  private def executeStatement(): Unit = withLocalProperties {
+  protected def incrementalCollectResult(): Iterator[Any] = {
+    result.toLocalIterator().asScala
+  }
+
+  protected def fullCollectResult(): Array[_] = {
+    result.collect()
+  }
+
+  protected def takeResult(maxRows: Int): Array[_] = {
+    result.take(maxRows)
+  }
+
+  protected def collectResult(): Unit = {
```

Comment:
why not pass result as parameter and return iter?

Reply:
updated